packages/issuer/src/blockchain-facade/Certificate.ts

```diff
@@ -468,4 +489,61 @@ export class Certificate extends PreciseProofEntity implements ICertificate {
 
         return revokedEvents.length > 0;
     }
+
+    async getClaimedData(): Promise<IClaim[]> {
+        const { registry } = this.configuration
+            .blockchainProperties as Configuration.BlockchainProperties<Registry, Issuer>;
+
+        const claims: IClaim[] = [];
+
+        const claimSingleEvents = await getEventsFromContract(
```

Comment:
This is fine for now, but seems there is no way to read the claim data without iterating over the events, super inefficient. We should discuss the improvement proposal on https://github.com/ethereum/EIPs/issues/1888